### ISSUE TITLE
multiparty account support

### ISSUE THREAD

**Issue by jl777** [lightningnetwork/lightning-rfc#220]
The current LN protocol appears to assume every address is a single party address. However, I believe with a small change, multisig addresses could be supported.

That would allow LN payments sent to a group controlled address and disbursement requiring signatures from MofN parties.

Not sure what changes other than having a multisig redeemscript is needed at the protocol level. At the implementation level, coordinating the MofN nodes would need to be solved, which might require opening up a signature round in the protocol.

I apologize that I am not better versed in the protocol to make more accurate suggestions.

**Comment by Roasbeef** [comment]
See [this post from the mailing list](https://lists.linuxfoundation.org/pipermail/lightning-dev/2016-January/000403.html) on how to achieve something like this.

**Comment by Roasbeef** [comment]
Actually, re-reading your initial message, are you referring to channels whose funds are explicitly controlled by multiple participants?

**Comment by jl777** [comment]
yes. I need an MofN signoff for releasing funds controlled by the group

**Comment by rustyrussell** [comment]
This is a big piece of work.  We don't have any plans to figure out all the corners and the protocol which would be required for the moment, but I'll leave this issue here as a reminder.
